scripts/apps/contacts/directives/ContactList.js

```diff
@@ -0,0 +1,217 @@
+ * @requires multi
+ * @requires Keys
+ * @requires gettextCatalog
+ * @requires $rootScope
+ * @requires config
+ * @requires metadata
+ * @requires storage
+ * @requires keyboardManager
+ *
+ * @description Handles the functionality displaying list of items from contacts collection
+ */
+
+export function ContactList(
+    $location,
+    $timeout,
+    $injector,
+    $filter,
+    search,
+    datetime,
+    gettext,
+    superdesk,
+    multi,
+    Keys,
+    gettextCatalog,
+    $rootScope,
+    config,
+    metadata,
+    storage,
+    keyboardManager
+) {
+    // contains all the injected services to be passed down to child
+    // components via props
+    const services = {
+        $location: $location,
+        $timeout: $timeout,
+        $injector: $injector,
+        $filter: $filter,
+        search: search,
+        datetime: datetime,
+        gettext: gettext,
+        superdesk: superdesk,
+        multi: multi,
+        Keys: Keys,
+        gettextCatalog: gettextCatalog,
+        $rootScope: $rootScope,
+        config: config,
+        metadata: metadata,
+        storage: storage,
+        keyboardManager: keyboardManager
+    };
+
+    return {
+        link: function(scope, elem) {
+            elem.attr('tabindex', 0);
+
+            var itemList = React.createElement(ItemListComponent,
+                angular.extend({
+                    svc: services,
+                    scope: scope
+                }));
+
+            var listComponent = ReactDOM.render(itemList, elem[0]);
+
+            scope.$watch('items', (items) => {
+                if (!items || !items._items) {
+                    return;
+                }
+
+                var itemsList = [];
+                var currentItems = {};
+                var itemsById = angular.extend({}, listComponent.state.itemsById);
+
+                items._items.forEach((item) => {
+                    var itemId = search.generateTrackByIdentifier(item);
+                    var oldItem = itemsById[itemId] || null;
+
+                    if (!oldItem || !_.isEqual(oldItem, item)) {
+                        itemsById[itemId] = angular.extend({}, oldItem, item);
+                    }
+
+                    if (!currentItems[itemId]) { // filter out possible duplicates
+                        currentItems[itemId] = true;
+                        itemsList.push(itemId);
+                    }
+                });
+
+                listComponent.setState({
+                    itemsList: itemsList,
+                    itemsById: itemsById,
+                    view: scope.view
+                }, () => {
+                    scope.rendering = scope.loading = false;
+                });
+            }, true);
+
+            scope.$watch('view', (newValue, oldValue) => {
+                if (newValue !== oldValue) {
+                    listComponent.setState({view: newValue});
+                }
+            });
+
+            scope.$on('item:update', (_e, data) => {
```

Comment:
There is no `item:update` notification from contacts endpoints. It will be different websocket notification.